types/index.d.ts

```diff
@@ -0,0 +1,261 @@
+// TypeScript Version: 2.3
+
+declare module 'react-google-maps' {
+    import { Component, ComponentClass, ReactElement, StatelessComponent } from 'react'
+
+    export interface WithGoogleMapProps {
+        containerElement: ReactElement<any>
+        mapElement: ReactElement<any>
+    }
+
+    export function withGoogleMap<P>(wrappedComponent: string | ComponentClass<P> | StatelessComponent<P>): ComponentClass<P & WithGoogleMapProps>
+
+    export interface GoogleMapProps {
```

Comment:
How do you generate these type definitions? By hand?

Could they be generate typed code from `src/macro` directly using `jscodeshift` API? That could save you lots of time.

Reply:
I do not see any documentation on how to do this. I could possibly spend time and write definitions generator that uses `jscodeshift` API but it will take much more time than writing these definitions. Also I'm sure that it will still require some manual adjustments.